### PR TITLE
Label definition datasource and resource added

### PR DESCRIPTION
**Pull request by roopesh83**

### CODE REVIEW

**Comment by loafoe** [comment]
@roopesh83 the main branch contains the updated `go-hsdp-api` reference in `go.mod`, rebasing your branch against the latest `main` branch of the provider should fix the compile errors in your tree.

**Comment by roopesh83** [comment]
@loafoe , currently my branch has the go-hsdp-api updated with the latest commits

File: docs/data-sources/cdl_label_definition.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+* `cdl_endpoint` - (Required) The CDL instance endpoint to query
+* `label_def_id` - (Required) The ID of the label definition to look up
+* `study_id`     - (Required) The ID of the Research Study which contains the label definition
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The GUID of the Label definitions
+* `label` -  The label name
```
Comment (loafoe) [inline docs/data-sources/cdl_label_definition.md:33]: `label_name` would be better here to distinguish it from the `labels` field

File: docs/data-sources/cdl_label_definition.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+* `label_scope` -	Use this parameter to specify for which CDL Data the LabelDefinition is applicable
+* `labels` - 	Use this parameter to specify your labels, or classes. Add one label for each class.
```
Comment (loafoe) [inline docs/data-sources/cdl_label_definition.md:35]: This seems to be well defined structure. Using JSON should only be used for free form structures like JSON Type where it is difficult to predict the depth and object structure. Looking at the example this should works as well no? Example:

```hcl

labels = ["good", "bad", "acceptable", "something", "something1"]

```

This is preferred over just a JSON string

File: docs/resources/cdl_label_definition.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  label_def_name = "TF create Test4"
+  description = "TF create desc"  
+  label_scope = <<labelScope
+  {
+        "type": "DataObject.DICOM"
+    }
+  labelScope
+  label = "videoQualityTF10"
+  type = "cdl/video-classification"
+  labels = <<labels
```
Comment (loafoe) [inline docs/resources/cdl_label_definition.md:20]: See comment above, this should just be a list of strings, unless there are other variations possible

File: docs/data-sources/cdl_label_definition.md (hunk 2 above)
Comment (roopesh83) [reply docs/data-sources/cdl_label_definition.md:35]: labels argument is now an array of string

File: docs/data-sources/cdl_label_definition.md (hunk 1 above)
Comment (roopesh83) [reply docs/data-sources/cdl_label_definition.md]: label -> label_name
(Changed name of arg)

File: docs/resources/cdl_label_definition.md (hunk 3 above)
Comment (roopesh83) [reply docs/resources/cdl_label_definition.md]: label scope is now a string, instead of json string

Comment (roopesh83) [reply docs/resources/cdl_label_definition.md]: Not using json string anymore for label definition args

**Review by loafoe — APPROVED** [review]

**Pull request merged**